.github/workflows/buildandpublish.yml

```
@@ -0,0 +1,31 @@
+name: Build and publish Ansible
+on:
+  push:
+    tags:
+      - v*
+jobs:
+  build-and-publish:
+    name: Build and publish Ansible to PyPI
+    runs-on: ubuntu-latest
+    steps:
```

This would be good to restrict too.
```suggestion
    permissions:
      contents: read

    steps:
```